src/ripple/basics/impl/ResolverAsio.cpp

```diff
@@ -194,7 +202,9 @@ class ResolverAsioImpl
         stop_async ();
 
         JLOG(m_journal.debug()) << "Waiting to stop";
-        m_stop_complete.wait();
+        std::unique_lock<std::mutex> lk{m_mut};
+        m_cv.wait(lk, [this]{return m_asyncHandlersCompleted;});
+        lk.unlock();
```

Comment:
Might be worth declaring the mutex here locally instead of a class member and making `m_asyncHandlersCompleted` atomic.